### ISSUE TITLE
Add option to log (failed) authentication attempts to enable the usage of tools like fail2ban and crowdsec

### ISSUE THREAD

**Issue by netw0rk-noob** [rustdesk/rustdesk-server#435]
**Is your feature request related to a problem? Please describe.**
Right now everyone can connect to the server repeatedly as often as he wants to try bruteforcing the pre-shared key which is needed to authenticate to the server. Besides that, denial-of-service attacks are possible. That is due to the fact that rustdesk does not log (failed) authentication attempts (with source ip addresses) anywhere which would make it possible to use tools like fail2ban or crowdsec to ban repeated offenders. 

**Describe the solution you'd like**
I'd like rustdesk-server to add an option to enable auth logging which would log (failed) authentication attempts to a logfile which could be used by above mentioned daemons to ban ip adresses which repeatedly fail to authenticate against the server. 

**Describe alternatives you've considered**
I did enable debug logs by adding `Environment="RUST_LOG=debug"` to the systemd unit which is starting rustdesk-server according to [the FAQ](https://github.com/rustdesk/rustdesk/wiki/FAQ#connection-log-on-server-side) but since that logs all connections and doesnt differentiate between successful/failed authentications using fail2ban on it creates the risk of banning legitimate users. That risk gets even greater when considering the fact that a successfully established remote control session logs 8 lines containing the source ip address while an illegitimate connection (trying to connect with a wrong key specified in the Client settings under `network` -> `ID/Relay server` -> `Key`) only logs 2 lines.

**Comment by ArtDmn** [comment]
It is definitely something that would be useful, I'm looking forward this exact same feature too!

**Comment by kramttocs** [comment]
Would be really nice to let Fail2Ban monitor rustdesk

**Comment by baudneo** [comment]
Throwing my vote in there to add a login failed line for crowdsec parsing

**Comment by baudneo** [comment]
Anything is better than NOTHING. Security is paramount for a project that sells itself on being a secure, convenient method to access your devices remotely.

**Comment by masterjuggler** [comment]
I just set up rustdesk for a company and also want to get it integrated with crowdsec and possibly SIEM. Please make it easier for us to handle security.

**Comment by realindiahotel** [comment]
How do we utilise this feature?

**Comment by kramttocs** [comment]
> How do we utilise this feature?

It just added logging for failed login attempts so how you utilize it is dependent upon if you are using anything like fail2ban or crowdsec to monitor the logs.

**Comment by FABNANCY** [comment]
Hellow the feature seem to be nice... 
But how to activate it on compose stack for exemple, and how to monitor it in logs ( wich moed, info, debug, trace).
Thanks.

**Comment by kramttocs** [comment]
Good question. I was waiting for the release but then forgot.
There is some info here: https://github.com/rustdesk/rustdesk/wiki/FAQ#access-logs

I am using docker and am not finding the logs in the file system (terminal). Figure it will just be a matter of mapping their folder to a local share but need to find them first.

**Comment by realindiahotel** [comment]
> Good question. I was waiting for the release but then forgot. There is some info here: https://github.com/rustdesk/rustdesk/wiki/FAQ#access-logs
> 
> I am using docker and am not finding the logs in the file system (terminal). Figure it will just be a matter of mapping their folder to a local share but need to find them first.

This is what I mean by how to use. Where are the logs to read? Is it in the same output that displays in the docker container stdout/stderr?

**Comment by kramttocs** [comment]
Yep, I mis-interpreted your earlier post. I would assume it's the same output and did a little more searching around in the container but never found the persisted log. 
Hopefully a maintainer will chime in and help point us in the right direction.

**Comment by realindiahotel** [comment]
There needs to be some instructions from developers just saying "we make log" is pretty poor
